webapp/channels/src/components/advanced_text_editor/use_priority.tsx

```diff
@@ -0,0 +1,168 @@
+                },
+            };
+        } else {
+            updatedDraft.metadata = {};
+        }
+
+        handleDraftChange(updatedDraft, {instant: true});
+        focusTextbox();
+    }, [focusTextbox, draft, handleDraftChange]);
+
+    const handlePostPriorityHide = useCallback(() => {
+        focusTextbox(true);
+    }, [focusTextbox]);
+
+    const handleRemovePriority = useCallback(() => {
+        handlePostPriorityApply();
+    }, [handlePostPriorityApply]);
+
+    const showPersistNotificationModal = useCallback((message: string, specialMentions: {[key: string]: boolean}, channelType: Channel['type'], onConfirm: () => void) => {
+        dispatch(openModal({
+            modalId: ModalIdentifiers.PERSIST_NOTIFICATION_CONFIRM_MODAL,
+            dialogType: PersistNotificationConfirmModal,
+            dialogProps: {
+                currentChannelTeammateUsername: channelTeammateUsername,
+                specialMentions,
+                channelType,
+                message,
+                onConfirm,
+            },
+        }));
+    }, [channelTeammateUsername, dispatch]);
+
+    const onSubmitCheck = useCallback((onConfirm: () => void) => {
+        if (
+            isPostPriorityEnabled &&
+            hasRequestedPersistentNotifications(draft?.metadata?.priority)
+        ) {
+            showPersistNotificationModal(draft.message, specialMentions, channelType, onConfirm);
+            return true;
+        }
+        return false;
+    }, [isPostPriorityEnabled, showPersistNotificationModal, draft, channelType, specialMentions]);
+
+    const labels = useMemo(() => (
+        (hasPrioritySet && !draft.rootId) ? (
+            <PriorityLabels
+                canRemove={!shouldShowPreview}
+                hasError={!isValidPersistentNotifications}
+                specialMentions={specialMentions}
+                onRemove={handleRemovePriority}
+                persistentNotifications={draft!.metadata!.priority?.persistent_notifications}
+                priority={draft!.metadata!.priority?.priority}
+                requestedAck={draft!.metadata!.priority?.requested_ack}
+            />
+        ) : undefined
+    ), [shouldShowPreview, draft, hasPrioritySet, isValidPersistentNotifications, specialMentions, handleRemovePriority]);
+
+    const additionalControl = useMemo(() =>
+        !draft.rootId && isPostPriorityEnabled && (
+            <PostPriorityPickerOverlay
+                key='post-priority-picker-key'
+                settings={draft.metadata?.priority}
+                onApply={handlePostPriorityApply}
+                onClose={handlePostPriorityHide}
+                disabled={shouldShowPreview}
+            />
+        ), [draft.rootId, isPostPriorityEnabled, draft.metadata?.priority, handlePostPriorityApply, handlePostPriorityHide, shouldShowPreview]);
+
+    return [labels, additionalControl, isValidPersistentNotifications, onSubmitCheck];
+};
```

Comment:
more of a nit picking, since the hooks can return object it would have read well since this contain return more than 2 params back for it to be an object. But thats just me